src/huggingface_hub/serialization/_torch.py

```diff
@@ -0,0 +1,197 @@
+# Copyright 2024 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains pytorch-specific helpers."""
+import importlib
+from functools import lru_cache
+from typing import TYPE_CHECKING, Dict, Tuple
+
+from ._base import FILENAME_PATTERN, MAX_SHARD_SIZE, StateDictSplit, split_state_dict_into_shards
+
+
+if TYPE_CHECKING:
+    import torch
```

Comment:
Shouldn't this be just `import torch`? I'm getting an error otherwise:
```
[...]

  File "/home/lysandre/Workspaces/python/huggingface_hub/src/huggingface_hub/serialization/_torch.py", line 181, in _get_dtype_size
    _float8_e4m3fn = getattr(torch, "float8_e4m3fn", None)
NameError: name 'torch' is not defined
```

Reply:
In `transformers` we have methods like `is_torch_available` to ensure we don't import missing things at runtime, is this the original idea with the `TYPE_CHECKING` here?